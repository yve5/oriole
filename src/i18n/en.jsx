const I18N_DICTIONARY = {
  LANG: 'en',
  LANGUAGE: 'English',
  DATE_FORMAT: 'MM/dd/yyyy',
  LOCALE: 'enUS',
  'About us': 'About us',
  'An error has been detected. Please try again.':
    'An error has been detected. Please try again.',
  Buttons: 'Buttons',
  Changelog: 'Changelog',
  Collapse: 'Collapse',
  Expand: 'Expand',
  Internationalization: 'Internationalization',
  Paging: 'Paging',
  'The request has been sent successfully.':
    'The request has been sent successfully.',
};

export default I18N_DICTIONARY;
