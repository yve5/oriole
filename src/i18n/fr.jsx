const I18N_DICTIONARY = {
  LANG: 'fr',
  LANGUAGE: 'Français',
  DATE_FORMAT: 'dd/MM/yyyy',
  LOCALE: 'fr',
  'About us': 'À propos de nous',
  'An error has been detected. Please try again.':
    'Une erreur a été détectée. Merci de recommencer.',
  Buttons: 'Boutons',
  Changelog: 'Journal des modifications',
  'Changes have been saved.': 'Les changements ont été sauvegardés.',
  'Choose a file': 'Choisir un fichier',
  Collapse: 'Réduire',
  Columns: 'Colonnes',
  'Creation date': 'Date de création',
  'Descending order': 'Ordre décroissant',
  Disabled: 'Désactivé',
  Enabled: 'Activé',
  Expand: 'Développer',
  First: 'Premier',
  'Hello World': 'Bonjour le monde',
  'Hide columns': 'Cacher les colonnes',
  Internationalization: 'Internationalisation',
  Paging: 'Pagination',
  'The request has been sent successfully.':
    'La demande a été envoyée avec succès.',
};

export default I18N_DICTIONARY;
