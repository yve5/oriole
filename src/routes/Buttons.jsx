import React, { Fragment } from 'react';
import Confetti from 'canvas-confetti';

const buttons = [
  { title: 'Primary', suffix: 'primary' },
  { title: 'Secondary', suffix: 'outline-secondary' },
  { title: 'Success', suffix: 'success' },
  { title: 'Danger', suffix: 'danger' },
  { title: 'Warning', suffix: 'warning' },
  { title: 'Info', suffix: 'info' },
  { title: 'Light', suffix: 'outline-light' },
  { title: 'Dark', suffix: 'dark' },
  { title: 'Link', suffix: 'link' },
  // { title: 'Outline Primary', suffix: 'outline-primary' },
  // { title: 'Outline Secondary', suffix: 'outline-secondary' },
  // { title: 'Outline Success', suffix: 'outline-success' },
  // { title: 'Outline Danger', suffix: 'outline-danger' },
  // { title: 'Outline Warning', suffix: 'outline-warning' },
  // { title: 'Outline Info', suffix: 'outline-info' },
  // { title: 'Outline Light', suffix: 'outline-light' },
  // { title: 'Outline Dark', suffix: 'outline-dark' },
  // { title: 'Outline Link', suffix: 'outline-link' },
];

const Buttons = () => (
  <div className="card">
    <ul className="list-group list-group-flush">
      <li className="list-group-item">
        {buttons.map(({ title, suffix }) => (
          <button
            className={`btn btn-${suffix} me-2 mb-2`}
            data-testid="button-demo"
            onClick={() =>
              Confetti({
                particleCount: 100,
                origin: { y: 0.6 },
                spread: 70,
              })
            }
            type="button"
            key={suffix}
          >
            {title}
          </button>
        ))}
      </li>
      <li className="list-group-item bg-light-subtle">
        <code>
          {buttons.map(({ title, suffix }) => (
            <Fragment key={suffix}>
              &lt;button className=&quot;btn btn-{suffix}&quot;
              type=&quot;button&quot;&gt;
              {title}&lt;/button&gt;
              <br />
            </Fragment>
          ))}
        </code>
      </li>
    </ul>
  </div>
);

export default Buttons;
