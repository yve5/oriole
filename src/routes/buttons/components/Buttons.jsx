import React, { Fragment } from 'react';
import Confetti from 'canvas-confetti';

import { ORIOLE_BUTTONS } from '../resources/constants';

const ButtonLayer = ({ className = '' }) => (
  <div className="card mb-4">
    <ul className="list-group list-group-flush">
      <li className="list-group-item">
        {ORIOLE_BUTTONS.map(({ title, suffix }) => (
          <button
            key={suffix}
            type="button"
            data-testid="button-demo"
            className={`btn btn-${suffix} me-2 mb-2 ${className}`}
            onClick={() =>
              Confetti({ spread: 70, origin: { y: 0.6 }, particleCount: 100 })
            }
          >
            {title}
          </button>
        ))}
      </li>
      <li className="list-group-item bg-light-subtle">
        <code>
          {ORIOLE_BUTTONS.map(({ title, suffix }) => (
            <Fragment key={suffix}>
              &lt;button className=&quot;btn btn-{suffix} {className}&quot;
              type=&quot;button&quot;&gt;
              {title}&lt;/button&gt;
              <br />
            </Fragment>
          ))}
        </code>
      </li>
    </ul>
  </div>
);

const Buttons = () => (
  <>
    <ButtonLayer className="btn-sm" />
    <ButtonLayer className="btn-sm disabled" />
  </>
);

export default Buttons;
