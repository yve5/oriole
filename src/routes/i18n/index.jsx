import I18nPage from './components/I18n';

export default I18nPage;
