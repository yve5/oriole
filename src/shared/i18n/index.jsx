import LangSwitch from './components/LangSwitch';
import I18n from './components/I18n';
import i18nReducer from './reducer';

export { I18n, LangSwitch, i18nReducer };
