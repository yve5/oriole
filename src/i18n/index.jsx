import fr from './fr';
import en from './en';
import { tsl } from '../shared/i18n/utils/Translate';

const i18n = {
  tsl: (term, args) => tsl(fr, term, args),
  dictionaries: [fr, en],
  lang: fr.LANG,
};

export default i18n;
