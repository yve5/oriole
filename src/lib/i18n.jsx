import { I18n, LangSwitch, i18nReducer } from '../shared/i18n';

export { I18n, LangSwitch, i18nReducer };
