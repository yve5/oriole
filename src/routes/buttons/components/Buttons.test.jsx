import React from 'react';
import configureMockStore from 'redux-mock-store';

import { thunk } from 'redux-thunk';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { act, fireEvent, render } from '@testing-library/react';

import Buttons from './Buttons';
import i18n from '../../../i18n';

const mockStore = configureMockStore([thunk]);

describe('Buttons', () => {
  it('render the component', async () => {
    const { asFragment, getAllByText } = render(
      <Provider store={mockStore({ i18n })}>
        <MemoryRouter>
          <Buttons />
        </MemoryRouter>
      </Provider>
    );

    await act(async () => {
      fireEvent.click(getAllByText('Primary')[0]);
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
