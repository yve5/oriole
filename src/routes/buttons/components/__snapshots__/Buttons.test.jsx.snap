// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Buttons > render the component 1`] = `
<DocumentFragment>
  <div
    class="card mb-4"
  >
    <ul
      class="list-group list-group-flush"
    >
      <li
        class="list-group-item"
      >
        <button
          class="btn btn-primary me-2 mb-2 btn-sm"
          data-testid="button-demo"
          type="button"
        >
          Primary
        </button>
        <button
          class="btn btn-secondary me-2 mb-2 btn-sm"
          data-testid="button-demo"
          type="button"
        >
          Secondary
        </button>
        <button
          class="btn btn-success me-2 mb-2 btn-sm"
          data-testid="button-demo"
          type="button"
        >
          Success
        </button>
        <button
          class="btn btn-danger me-2 mb-2 btn-sm"
          data-testid="button-demo"
          type="button"
        >
          Danger
        </button>
        <button
          class="btn btn-warning me-2 mb-2 btn-sm"
          data-testid="button-demo"
          type="button"
        >
          Warning
        </button>
        <button
          class="btn btn-info me-2 mb-2 btn-sm"
          data-testid="button-demo"
          type="button"
        >
          Info
        </button>
        <button
          class="btn btn-link me-2 mb-2 btn-sm"
          data-testid="button-demo"
          type="button"
        >
          Link
        </button>
      </li>
      <li
        class="list-group-item bg-light-subtle"
      >
        <code>
          &lt;button className="btn btn-primary btn-sm" type="button"&gt;Primary&lt;/button&gt;
          <br />
          &lt;button className="btn btn-secondary btn-sm" type="button"&gt;Secondary&lt;/button&gt;
          <br />
          &lt;button className="btn btn-success btn-sm" type="button"&gt;Success&lt;/button&gt;
          <br />
          &lt;button className="btn btn-danger btn-sm" type="button"&gt;Danger&lt;/button&gt;
          <br />
          &lt;button className="btn btn-warning btn-sm" type="button"&gt;Warning&lt;/button&gt;
          <br />
          &lt;button className="btn btn-info btn-sm" type="button"&gt;Info&lt;/button&gt;
          <br />
          &lt;button className="btn btn-link btn-sm" type="button"&gt;Link&lt;/button&gt;
          <br />
        </code>
      </li>
    </ul>
  </div>
  <div
    class="card mb-4"
  >
    <ul
      class="list-group list-group-flush"
    >
      <li
        class="list-group-item"
      >
        <button
          class="btn btn-primary me-2 mb-2 btn-sm disabled"
          data-testid="button-demo"
          type="button"
        >
          Primary
        </button>
        <button
          class="btn btn-secondary me-2 mb-2 btn-sm disabled"
          data-testid="button-demo"
          type="button"
        >
          Secondary
        </button>
        <button
          class="btn btn-success me-2 mb-2 btn-sm disabled"
          data-testid="button-demo"
          type="button"
        >
          Success
        </button>
        <button
          class="btn btn-danger me-2 mb-2 btn-sm disabled"
          data-testid="button-demo"
          type="button"
        >
          Danger
        </button>
        <button
          class="btn btn-warning me-2 mb-2 btn-sm disabled"
          data-testid="button-demo"
          type="button"
        >
          Warning
        </button>
        <button
          class="btn btn-info me-2 mb-2 btn-sm disabled"
          data-testid="button-demo"
          type="button"
        >
          Info
        </button>
        <button
          class="btn btn-link me-2 mb-2 btn-sm disabled"
          data-testid="button-demo"
          type="button"
        >
          Link
        </button>
      </li>
      <li
        class="list-group-item bg-light-subtle"
      >
        <code>
          &lt;button className="btn btn-primary btn-sm disabled" type="button"&gt;Primary&lt;/button&gt;
          <br />
          &lt;button className="btn btn-secondary btn-sm disabled" type="button"&gt;Secondary&lt;/button&gt;
          <br />
          &lt;button className="btn btn-success btn-sm disabled" type="button"&gt;Success&lt;/button&gt;
          <br />
          &lt;button className="btn btn-danger btn-sm disabled" type="button"&gt;Danger&lt;/button&gt;
          <br />
          &lt;button className="btn btn-warning btn-sm disabled" type="button"&gt;Warning&lt;/button&gt;
          <br />
          &lt;button className="btn btn-info btn-sm disabled" type="button"&gt;Info&lt;/button&gt;
          <br />
          &lt;button className="btn btn-link btn-sm disabled" type="button"&gt;Link&lt;/button&gt;
          <br />
        </code>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
